python/ray/data/block.py

```diff
@@ -222,6 +227,38 @@ def __post_init__(self):
             self.input_files = []
 
 
+@DeveloperAPI
+@dataclass
+class BlockMetadataWithSchema(BlockMetadata):
```

Comment:
Let's add an assertion into `DataOpTask.on_data_ready` that this class is never returned from the launched tasks (so that someone wouldn't start passing schemas inside metadata again)